ulog.cpp

```diff
@@ -0,0 +1,316 @@
+/*
+ * This file is part of the MAVLink Router project
+ *
+ * Copyright (C) 2017  Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "ulog.h"
+
+#include <assert.h>
+
+#include "log.h"
+#include "util.h"
+
+#define ULOG_HEADER_SIZE 16
+#define ULOG_MAGIC                               \
+    {                                            \
+        0x55, 0x4C, 0x6F, 0x67, 0x01, 0x12, 0x35 \
+    }
+
+#define NO_FIRST_MSG_OFFSET 255
+
+struct _packed_ ulog_msg_header {
+    uint16_t msg_size;
+    uint8_t msg_type;
+};
+
+ULog::ULog(const char *logs_dir, Timeout_Manager *timeout_manager)
+    : Endpoint{"ULog", false}
+{
+    assert(timeout_manager);
+    assert(logs_dir);
+
+    _timeout_manager = timeout_manager;
+    _logs_dir = strdup(logs_dir);
```

Comment:
Oh, and here it's dupped again - no need to it. Believe in `const char*` magic =D